submissions/MosQuitO404/OOPJS_frogger/app.js

```diff
@@ -0,0 +1,131 @@
+// Enemies our player must avoid
+const Enemy = function() {
+    // Variables applied to each of our instances go here,
+    // we've provided one for you to get started
+    
+    // The image/sprite for our enemies, this uses
+    // a helper we've provided to easily load images
+    this.x = -95;
+    this.y = this.positionEnemyMaker();
+    this.speed = this.speedEnemyMaker(200, 450);
+    this.sprite = 'images/enemy-bug.png';
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function(dt) {
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+    // all computers.
+    if (this.x < 500) { 
+        this.x += Math.round(this.speed * dt);
+    }
+    else {
+        this.x = -95;
+        this.y = this.positionEnemyMaker();
+        this.speed = this.speedEnemyMaker(200, 450);
+    }
+    this.checkCollisions();
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.positionEnemyMaker = function() {
+    const allowedPositions = [47, 129, 211]; 
+    return allowedPositions[Math.floor(Math.random() * allowedPositions.length)]; 
+}
+
+Enemy.prototype.speedEnemyMaker = function(maxSpeed, minSpeed) {
+    return Math.floor(Math.random() * (maxSpeed - minSpeed)) + minSpeed; 
+}
+
+Enemy.prototype.checkCollisions = function() {
+    if (player.y < this.y + player.collisionAreaHeight && 
+        player.y + player.collisionAreaHeight > this.y &&
+        player.x < this.x + player.collisionAreaWidth && 
+        player.x + player.collisionAreaWidth > this.x) {
+            player.x = player.columnWidth * 2; 
+            player.y = player.rowHeight * 5;
+    }
+};
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+const Player = function() {
+    this.rowHeight = 75;
+    this.columnWidth = 101;
```

Comment:
These do not seem to be properties of a player. Player doesn't have rows or columns.

Reply:
Ok! Done